src/scales/scale.timeseries.js

```diff
@@ -0,0 +1,190 @@
+			unit: false, // false == automatic or override with week, month, year, etc.
+			round: false, // none, or override with week, month, year, etc.
+			displayFormat: false, // DEPRECATED
+			minUnit: 'millisecond',
+
+			// defaults to unit's corresponding unitFormat below or override using pattern string from http://momentjs.com/docs/#/displaying/format/
+			displayFormats: {
+				millisecond: 'h:mm:ss.SSS a', // 11:20:01.123 AM,
+				second: 'h:mm:ss a', // 11:20:01 AM
+				minute: 'h:mm:ss a', // 11:20:01 AM
+				hour: 'MMM D, hA', // Sept 4, 5PM
+				day: 'll', // Sep 4 2015
+				week: 'll', // Week 46, or maybe "[W]WW - YYYY" ?
+				month: 'MMM YYYY', // Sept 2015
+				quarter: '[Q]Q - YYYY', // Q3
+				year: 'YYYY' // 2015
+			},
+		},
+		ticks: {
+			autoSkip: true
+		}
+	};
+
+	function arrayUnique(arr) {
+		var result = [];
+		for (var i = 0; i < arr.length; i++) {
+			if (result.indexOf(arr[i]) === -1) {
+				result.push(arr[i]);
+			}
+		}
+		return result;
+	}
+
+	var TimeSeriesScale = Chart.TimeScaleBase.extend({
+		/**
+		* Internal function to get the correct labels. If data.xLabels or data.yLabels are defined, use those
+		* else fall back to data.labels
+		* @private
+		*/
+		getLabels: function() {
+			var data = this.chart.data;
+			return data.xLabels || data.labels;
+		},
+
+		determineDataLimits: function() {
+			var me = this;
+			var timeOpts = me.options.time;
+
+			var chartData = me.chart.data;
+			var parsedData = {
+				labels: [],
+				datasets: []
+			};
+
+			helpers.each(me.getLabels(), function(label, labelIndex) {
+				var labelMoment = timeHelpers.parseTime(me, label);
+
+				if (labelMoment.isValid()) {
+					// We need to round the time
+					if (timeOpts.round) {
+						labelMoment.startOf(timeOpts.round);
+					}
+					// Store this value for later
+					parsedData.labels[labelIndex] = labelMoment.valueOf();
+				}
+			});
+
+			helpers.each(chartData.datasets, function(dataset, datasetIndex) {
+				var timestamps = [];
+
+				if (typeof dataset.data[0] === 'object' && dataset.data[0] !== null && me.chart.isDatasetVisible(datasetIndex)) {
+					// We have potential point data, so we need to parse this
+					helpers.each(dataset.data, function(value, dataIndex) {
+						var dataMoment = timeHelpers.parseTime(me, me.getRightValue(value));
+
+						if (dataMoment.isValid()) {
+							if (timeOpts.round) {
+								dataMoment.startOf(timeOpts.round);
+							}
+
+							timestamps[dataIndex] = dataMoment.valueOf();
+						}
+					});
+				} else {
+					// We have no x coordinates, so use the ones from the labels
+					timestamps = parsedData.labels.slice();
+				}
+
+				parsedData.datasets[datasetIndex] = timestamps;
+			});
+
+			var allTimestamps = parsedData.labels;
+			helpers.each(parsedData.datasets, function(value) {
+				allTimestamps = allTimestamps.concat(value);
+			});
+
+			allTimestamps = arrayUnique(allTimestamps).sort(function(a, b) {
```

Comment:
I'm not sure we should sort here. I would expect the user to pass the data points in sorted already. If they want to plot in reverse chronological order we should probably let them do that, but sorting would stop that from working